src/connectors/tests/bench.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright 2022, The Tremor Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing perm
+use super::ConnectorHarness;
+use crate::{
+    connectors::{impls::bench, sink::SinkMsg},
+    errors::Result,
+    system::{World, WorldConfig},
+};
+use std::{
+    io::Write,
+    time::{Duration, Instant},
+};
+use tempfile::NamedTempFile;
+use tremor_common::ports::IN;
+use tremor_value::prelude::*;
+
+#[async_std::test]
+async fn stop_after_events() -> Result<()> {
+    let _ = env_logger::try_init();
+
+    let mut file = NamedTempFile::new()?;
```

Comment:
instead of creating a new file, could we use a fixture here?

Reply:
We could, but i didnt like to depend on the current working directory while executing tests...

Reply:
I don't like creating files :D bu I think it comes down to "like" this is your PR so your like matters!